client/src/models/Task.js

```diff
@@ -153,11 +178,75 @@ export default class Task extends Model {
     super(Model.fillObject(props, Task.yupSchema))
   }
 
+  isTopLevelTask() {
+    return _isEmpty(this.customFieldRef1)
+  }
+
+  fieldSettings() {
+    return this.isTopLevelTask()
+      ? Settings.fields.task.topLevel
+      : Settings.fields.task.subLevel
+  }
+
+  periodAssessmentYupSchema() {
+    return this.isTopLevelTask()
+      ? Task.topLevelAssessmentCustomFieldsSchema
+      : Task.subLevelAssessmentCustomFieldsSchema
+  }
+
+  periodAssessmentConfig() {
+    return this.fieldSettings().assessment?.customFields
+  }
+
   iconUrl() {
     return TASKS_ICON
   }
 
   toString() {
     return `${this.shortName}`
   }
+
+  getAssessmentResults(dateRange) {
+    const publishedReportsUuids = this.publishedReports.map(r => r.uuid)
+    const taskAssessmentNotes = this.notes
+      .filter(
+        n =>
+          n.type === NOTE_TYPE.ASSESSMENT &&
+          n.noteRelatedObjects.length === 2 &&
+          n.noteRelatedObjects.filter(
+            ro =>
+              ro.relatedObjectType === Report.relatedObjectType &&
+              publishedReportsUuids.includes(ro.relatedObjectUuid)
+          ).length &&
+          (!dateRange ||
+            (n.createdAt <= dateRange.end && n.createdAt >= dateRange.start))
+      )
+      .map(ta => JSON.parse(ta.text))
+    const assessmentResults = {}
+    taskAssessmentNotes.forEach(o =>
+      Object.keys(o).forEach(k => {
+        assessmentResults[k] = assessmentResults[k] || []
+        assessmentResults[k].push(o[k])
+      })
+    )
+    return assessmentResults
+  }
+
+  getLastAssessment(dateRange) {
+    const notesToAssessments = this.notes
+      .filter(n => {
+        return (
+          n.type === NOTE_TYPE.ASSESSMENT &&
+          n.noteRelatedObjects.length === 1 &&
+          (!dateRange ||
+            (n.createdAt <= dateRange.end && n.createdAt >= dateRange.start))
+        )
+      })
+      .sort((a, b) => b.createdAt - a.createdAt) // desc sorted
+      .map(ta => ({
+        uuid: ta.uuid,
+        assessment: JSON.parse(ta.text)
+      }))
+    return notesToAssessments.length ? notesToAssessments[0].assessment : null
```

Comment:
Could use `return notesToAssessments?.[0]?.assessment`

Reply:
Will also do it as part of the implementation of #2954